### PR TITLE
Bump hashicorp vault to 1.4.0.

### PR DESCRIPTION
**Pull request by dlorenc**
Replaces #296.

Signed-off-by: Dan Lorenc <lorenc.d@gmail.com>

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

File: .github/workflows/e2e_test.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ on:
     branches: [ main ]
 
 env:
-  GO_VERSION: 1.16.8
+  GO_VERSION: 1.17.3
```
Comment (cpanato) [inline .github/workflows/e2e_test.yml:25]: update to `1.17.7`

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ on:
     branches: [ main ]
 
 env:
-  GO_VERSION: 1.16.8
+  GO_VERSION: 1.17.3
```
Comment (cpanato) [inline .github/workflows/test.yml:28]: update to `1.17.7`

File: .github/workflows/e2e_test.yml (hunk 1 above)
Comment (dlorenc) [reply .github/workflows/e2e_test.yml]: Thanks!

**Comment by dlorenc** [comment]
Thanks, updated!

**Review by cpanato — APPROVED** [review]

**Pull request merged**